app/debt/check-debts.js

```diff
@@ -1,23 +1,34 @@
 const db = require('../data')
 
-const checkDebts = async (schemeId, frn, reference, netValue, transaction) => {
+const checkDebts = async (schemeId, frn, reference, secondaryReference, netValue, transaction) => {
   const parsedFrn = parseInt(frn)
 
   if (isNaN(parsedFrn)) {
     return {}
   } else {
     const referenceNumeric = reference.match(/\d+/)[0]
-
+    const secondaryReferenceNumeric = secondaryReference.match(/\d+/)[0]
     return db.debtData.findOne({
       transaction,
       where: {
         schemeId,
         frn: parsedFrn,
         netValue,
         paymentRequestId: null,
-        reference: {
-          [db.Sequelize.Op.like]: `%${referenceNumeric}`
-        }
+        [db.Sequelize.Op.or]: [
+          { reference },
+          { reference: secondaryReference },
+          {
+            reference: {
+              [db.Sequelize.Op.like]: `%${Number(referenceNumeric).toString()}%`
```

Comment:
Concern with this approach is if they entered just a number `1` as the reference, would any payment request with a `1` in it be viable for selection?

Reply:
Yeah, I suppose it would. What about just 

`[db.Sequelize.Op.like]: `%${Number(referenceNumeric).toString()}``

So we remove the final wildcard so it needs to be XYZreference and things like XYZreferenceABC are not caught as a match?

Reply:
That's better, but if they still just put `1` then anything ending `1` will be a potential target.

Could we somehow make this CS specific vs not CS?

Maybe a minimum amount of characters the user must input too?

Reply:
Yeah, I suppose we could do CS specific to cut down on how often we do these extra checks and open ourselves up.

Do you reckon just CS, change to the wildcards is enough? I'm mindful we're also filtering on the netValue and the schemeId AND the FRN already, so we are adding failsafes for (hopefully) unlikely circumstance

Reply:
If only users had access to the invoice number and this whole thing would be a piece of cake.

You make a good point though, we're narrowing the scope of potential targets down already.

I think we could keep them all consistent with % at start but specify a minimum number of characters the user must input to really minimise the risk.